lib/nerves_hub/deployments.ex

```diff
@@ -352,8 +354,9 @@ defmodule NervesHub.Deployments do
       )
       |> where([d], d.firmware_metadata["platform"] == ^deployment.firmware.platform)
       |> where([d], d.firmware_metadata["architecture"] == ^deployment.firmware.architecture)
+      |> where([d], fragment("? <@ ?", ^deployment.conditions["tags"], d.tags))
 
-    Repo.insert_all(InflightDeploymentCheck, query)
+    Repo.insert_all(Oban.Job, query)
```

Comment:
@nshoes regarding the timeouts you were seeing, we could change this to 
```suggestion
    Repo.insert_all(Oban.Job, query, timeout: 30_000)
```
and see if that helps?